### ISSUE TITLE
spaCy init fill-config CLI throws no such option -c error for custom functions

### ISSUE THREAD

**Issue by AashishTiwari** [explosion/spaCy#7539]
## How to reproduce the behaviour
```
python -m spacy init fill-config some_base_config.cfg config.cfg -c scripts/custom_functions.py
```

<details>
<summary>Error Message</summary>

```
Usage: python -m spacy init fill-config [OPTIONS] BASE_PATH [OUTPUT_FILE]
Try 'python -m spacy init fill-config --help' for help.

Error: no such option: -c
```

</details>
 

Similarly for other command line args such as "--code" or "--code-path"
Link to Documentation: [Fill Config](https://spacy.io/api/cli#init-fill-config)

## Help Docs

<details>
<summary>Help Output</summary>

```
python -m spacy init fill-config --help

```
Usage: python -m spacy init fill-config [OPTIONS] BASE_PATH [OUTPUT_FILE]

  Fill partial config.cfg with default values. Will add all missing settings
  from the default config and will create all objects, check the registered
  functions for their default values and update the base config. This
  command can be used with a config generated via the training quickstart
  widget: https://spacy.io/usage/training#quickstart

  DOCS: https://spacy.io/api/cli#init-fill-config

Arguments:
  BASE_PATH      Base config to fill  [required]
  [OUTPUT_FILE]  File to save config.cfg to (or - for stdout)  [default: -]

Options:
  -pt, --pretraining  Include config for pretraining (with 'spacy pretrain')
                      [default: False]

  -D, --diff          Print a visual diff highlighting the changes  [default:
                      False]

  --help              Show this message and exit.

</details>

## Info about spaCy

- **spaCy version:** 3.0.5
- **Platform:** Linux-5.4.0-1039-aws-x86_64-with-debian-buster-sid
- **Python version:** 3.6.13
- **Pipelines:** en_core_web_trf (3.0.0)

Additionally, it will be good to know other suggested methods to register custom architectures for new projects. 
Thanks!

**Comment by adrianeboyd** [comment]
Hi, yes, this is a bug in v3.0.5. It was just fixed in #7421 and should work in the upcoming v3.0.6.

**Comment by AashishTiwari** [comment]
Hi @adrianeboyd, Thanks for this pointer, I applied the mentioned patch and it resolves the issue. Please close the bug report as needed, since it will be fixed via upcoming v3.0.6. 
Thanks!

**Comment by svlandeg** [comment]
Thanks for reporting back, @AashishTiwari !
